R/update_definition.R

```diff
@@ -0,0 +1,117 @@
+#' Update Feature Layer or Service definition
+#'
+#' <https://developers.arcgis.com/rest/services-reference/online/update-definition-feature-layer-.htm#ESRI_SECTION2_027B8A04953C409E9A9D9051DD4CDBEE>
+#' <https://developers.arcgis.com/rest/services-reference/online/update-definition-feature-service-.htm>
+#'
+#' @param x A Feature Layer or Feature Service class object.
+#' @param update A named list used as the "updateDefinition" body for the
+#'   request.
+#' @param async Default `FALSE`. If `TRUE`, support asynchronous processing for
+#'   the request.
+#' @export
+update_definition <- function(
+    x,
+    update = NULL,
```

Comment:
I think this is a really good start! I'm thinking we should probably use `...` here personally. That way we can `rlang::list2()` and then `lapply()` over the list to cast to json for the body